truenas_install/fhs.py

```diff
@@ -0,0 +1,105 @@
+"""
+TrueNAS Filesysten Heirarchy Standards extensions
+
+The TRUENAS_DATASET dictionary contains the dataset configuration
+settings for the root filesystem of the truenas server.
+
+When practical it is best to turn off unnecessary features.
+For example, disabling ACL support on system files simplifies
+permissions auditing. Disabling ACL support also ensures compliance
+with STIGs that require configuration files and libraries to not have ACLs
+present.
+
+KEYS
+--------------
+The following keys are supported:
+`name` - the name of the dataset (will be appended to other dataset
+         name related components
+`options` - Dataset configuration options (explained below)
+`mode` - permissions to set on the dataset's mountpoint during installation
+`mp` - dataset mountpoint.
+`snap` - Take a snapshot named "pristine" after creating the dataset.
+
+OPTIONS
+--------------
+NOSUID - sets a combination of setuid=off and devices=off
+NOEXEC - sets exec=off
+NOACL - sets acltype=off.
+NOATIME - sets atime=off.
+RO - sets readonly=on.
+NODEV - sets devices=off
+
+DATASETS
+--------------
+audit - dataset used for storing system auditing databases
+conf - truenas configuration files - static
+data - truenas configuraiton files - dynamic
+etc - see FHS
+home - see FHS, NOTE: only `admin` account will be present
+mnt - TrueNAS does not follow FHS for this path. It is reserved exclusively
+    for ZFS pool mountpoints.
+opt - see FHS
+usr - see FHS
+var - see FHS
+var/log - separate dataset for system logs. This is to provide flexibility to
+    snapshot and replicate log information as needed by administrator.
+var/ca-certificates - administrator-provided CA certificates, symlinked
+    from /usr/local/share/ca-certificates.
+"""
+
+TRUENAS_DATASETS = [
```

Comment:
We can probably use json schema validation here to ensure that this is valid and the github CI action can catch that (can be done separately too in a ticket i guess :P)

Reply:
Also we can set default variable values based on the same schema

Reply:
No I don't agree, we are actively trying to remove dependencies on 3rd party tools and I definitely don't want to add this dependency to our installer. It's overkill for what we want to do. We can revisit this request in another ticket but for now, I don't want to do this.

Reply:
@yocalebo json schema is already a dependency of scale-build

Reply:
It's not a dependency in our installer though.

Reply:
Okay. I created a schema and plumbed into our scale-build validate methods (rather than doing at runtime).